src/middleware/language-switcher.ts

```diff
@@ -26,7 +27,7 @@ export const localeUrl = (
   // }
 
   const newPath = isEmpty(pathElements) ? '' : `/${pathElements.join('/')}`;
-  const queryString = isEmpty(query) ? '' : `?${new URLSearchParams(query).toString()}`;
+  const queryString = isEmpty(query) ? '' : `?${qs.stringify(query)}`;
```

Comment:
what does qs do that URLSearchParams doesn't?

Reply:
preserves nested structures, else you ended up with things like filters[].a=1,filters[].b=2 etc (it breaks)